src/SAML2/XML/AbstractXMLElement.php

```diff
@@ -60,6 +67,29 @@ public function unserialize($serialized): void
     }
 
 
+    /**
+     * Create a document structure for this element
+     *
+     * @param \DOMElement|null $parent The element we should append to.
+     * @return \DOMElement
+     */
+    public function instantiateParentElement(DOMElement $parent = null): DOMElement
+    {
+        $qualifiedName = join('', array_slice(explode('\\', get_class()), -1));
```

Comment:
I think this is wrong. Calling `get_class()` without params assumes the current class, which in this case is `SAML2\XML\AbstractXMLElement`. We should either be using `get_called_class()` or `static::class`, since that will use late static binding and therefore point to the right class.

Reply:
Just to demonstrate it:

```php
php > class A {
php {     public static function m() {
php {         echo get_class();
php {     }
php { }
php > class B extends A {}
php > echo A::m();
A
php > echo B::m();
A
```

Reply:
Did I fix it wrong? Or would my solution work as well?

Reply:
Both are equivalent, yep. I just didn't see your fix while I was taking a look at the PR 😅